nixos/modules/services/web-apps/birdsitelive.nix

```diff
@@ -0,0 +1,448 @@
+                  '';
+                };
+              };
+
+              Moderation = {
+                FollowersWhiteListing = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Fediverse user or instance patterns to allow as followers.";
+                  apply = concatOrNull;
+                };
+
+                FollowersBlackListing = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Fediverse user or instance patterns to disallow as followers.";
+                  apply = concatOrNull;
+                };
+
+                TwitterAccountsWhiteListing = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Twitter handles to allow following.";
+                  apply = concatOrNull;
+                };
+
+                TwitterAccountsBlackListing = mkOption {
+                  type = with types; listOf str;
+                  default = [ ];
+                  description = mdDoc "List of Twitter handles to disallow following.";
+                  apply = concatOrNull;
+                };
+              };
+            };
+          };
+        };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    warnings = let mod = cfg.settings.Moderation; in
+      lib.optional (mod.FollowersWhiteListing != null && mod.FollowersBlackListing != null ||
+        mod.TwitterAccountsWhiteListing != null && mod.TwitterAccountsBlackListing != null) ''
+        If both both a whitelist and a blacklist are set for Fediverse or Twitter
+        accounts, only the whitelist will be used.
+      '';
+
+    users = {
+      users."${cfg.user}" = {
+        description = "BirdsiteLIVE user";
+        group = cfg.group;
+        isSystemUser = true;
+      };
+      groups."${cfg.group}" = { };
+    };
+
+    # Confinement of the main service unit requires separation of the
+    # configuration generation into a separate unit to permit access to secrets
+    # residing outside of the chroot.
+    systemd.services.birdsitelive-config = {
+      description = "BirdsiteLIVE Twitter bridge configuration";
+      reloadTriggers = [ configFile ] ++ secretPaths;
+
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+
+        RuntimeDirectory = "birdsitelive";
+        RuntimeDirectoryMode = "0750";
+
+        ExecStart = [
+          reloadScript
+          "-${managerReloadScript}"
+        ];
+        ExecReload = [
+          reloadScript
+          "-${managerReloadScript}"
+        ];
+      };
+    };
+
+    systemd.services.birdsitelive = {
+      confinement = {
+        enable = true;
+        packages = with pkgs; [ dotnetCorePackages.aspnetcore_3_1 ];
+      };
```

Comment:
We don't use `confinement` in NixOS modules. There has never been consensus to use this and it is a very large shift that would require agreement to start pushing out in our modules. Please remove.

Reply:
Would it be sufficient to just disable it by default?

Reply:
We could do that but there is also the chance that it bit rots very fast.

Reply:
Presumably the author will run the module with confinement enabled so it might not be so bad...

Reply:
That is my intention. I also made sure that the module test runs both with and without confinement enabled.